src/n-acd.c

```diff
@@ -296,7 +296,7 @@ int n_acd_ensure_bpf_map_space(NAcd *acd) {
 
         r = n_acd_bpf_compile(&fd_prog, fd_map, (struct ether_addr*) acd->mac);
         if (r)
-                return r;
+                fd_prog = -1;
```

Comment:
This will suppress any real error from `n_acd_bpf_compile()`. Why not add `N_ACD_E_DENIED` after `N_ACD_E_DROPPED` in `n-acd-private.h` and make `n_acd_bpf_compile()` return that in the exact circumstances where it matters? Then make all callers of it handle that error gracefully, just like you did here.